src/services/teams/hooks/index.js

```diff
@@ -377,47 +375,60 @@ exports.teamRolesToHook = teamRolesToHook;
  * @return {function::function(hook)}
  * @ifNotLocal
  */
-const hasTeamPermission = (permsissions, _teamId) => {
-	return globalHooks.ifNotLocal((hook) => {
-
+const hasTeamPermission = (permsissions, _teamId) => globalHooks.ifNotLocal((hook) => {
+	if (get(hook, 'isSuperhero') === true) {
+		return hook;
+	}
+	if (isString(permsissions)) {
+		// eslint-disable-next-line no-param-reassign
+		permsissions = [permsissions];
+	}
+	return Promise.all(
+		[getSessionUser(hook), teamRolesToHook(hook), getTeam(hook)],
+	).then(([ref, team]) => {
 		if (get(hook, 'isSuperhero') === true) {
 			return hook;
 		}
-		if (isString(permsissions)) {
-			permsissions = [permsissions];
-		}
-		return Promise.all(
-			[getSessionUser(hook), teamRolesToHook(hook), getTeam(hook)],
-		).then(([sessionUser, ref, team]) => {
-			if (get(hook, 'isSuperhero') === true) {
-				return hook;
-			}
-			const userId = bsonIdToString(hook.params.account.userId);
-			const teamId = _teamId || hook.teamId || hook.id;
-			const teamUser = team.userIds.find(_user => isSameId(_user.userId, userId));
+		// const roleRole = sessionUser.roles.filter(role => role.name === 'student');
+		// if (roleRole) {
+		//	throw new Forbidden('Students are not allowed to invite Extern!');
+		// }
+		const userId = bsonIdToString(hook.params.account.userId);
+		const teamId = _teamId || hook.teamId || hook.id;
+		const teamUser = team.userIds.find(_user => isSameId(_user.userId, userId));
 
-			if (isUndefined(teamUser)) {
-				throw new NotFound(`Session user is not in this team userId=${userId} teamId=${teamId}`);
-			}
+		if (isUndefined(teamUser)) {
+			throw new NotFound(`Session user is not in this team userId=${userId} teamId=${teamId}`);
+		}
 
-			const teamRoleId = teamUser.role;
-			const userTeamPermissions = ref.findRole('_id', teamRoleId, 'permissions');
+		const teamRoleId = teamUser.role;
+		const userTeamPermissions = ref.findRole('_id', teamRoleId, 'permissions');
 
-			permsissions.forEach((_permsission) => {
-				if (userTeamPermissions.includes(_permsission) === false) {
-					throw new Forbidden(`No permission=${_permsission} found!`);
-				}
-			});
-
-			return Promise.resolve(hook);
-		}).catch((err) => {
-			logger.warn(err);
-			throw new Forbidden('You have not the permission to access this. (2)');
+		permsissions.forEach((_permsission) => {
+			if (userTeamPermissions.includes(_permsission) === false) {
+				throw new Forbidden(`No permission=${_permsission} found!`);
+			}
 		});
+
+		return Promise.resolve(hook);
+	}).catch((err) => {
+		logger.warn(err);
+		throw new Forbidden('You have not the permission to access this. (2)');
 	});
-};
+});
 exports.hasTeamPermission = hasTeamPermission; // to use it global
 
+
+const denyRoleToInviteExtern = userRole => globalHooks.ifNotLocal(hook => Promise.all(
```

Comment:
this is a generic file and should not contain specific methods. 
i would prefer a method name like denyRole(role, ...)

Reply:
it is already in teams/hooks scope.